core/src/main/scala/org/apache/spark/util/DependencyUtils.scala

```diff
@@ -15,22 +15,112 @@
  * limitations under the License.
  */
 
-package org.apache.spark.deploy
+package org.apache.spark.util
 
 import java.io.File
-import java.net.URI
+import java.net.{URI, URISyntaxException}
 
 import org.apache.commons.lang3.StringUtils
 import org.apache.hadoop.conf.Configuration
 import org.apache.hadoop.fs.{FileSystem, Path}
 
 import org.apache.spark.{SecurityManager, SparkConf, SparkException}
+import org.apache.spark.deploy.SparkSubmitUtils
 import org.apache.spark.internal.Logging
-import org.apache.spark.util.{MutableURLClassLoader, Utils}
 
-private[deploy] object DependencyUtils extends Logging {
+private[spark] object DependencyUtils extends Logging {
+
+  def getIvyProperties(): Seq[String] = {
```

Comment:
`Seq[String]` seems like an odd return value for this. Maybe a tuple would be more appropriate? Or even a case class? For either seq or tuple I think we need to add some Scaladoc to tell callers how to unpack it.

Also I wonder if we should be returning `Option[String]` instead of `null` for missing values?

Reply:
add case class is good and then we don't need docs since parameter name is clear enough.
But change to Option[String] will change a lot related function, I think we can refactor this later like  https://github.com/apache/spark/pull/29966#discussion_r531917374

Reply:
Sounds good.